Makefile

```diff
@@ -150,10 +150,21 @@ typescript: prepare_lib
 	# Copy generated files to the frontend
 	mkdir -p frontend/src/Generated
 	cp $(TS_OUT)/segmented-cfmm/src/common.ts frontend/src/Generated/common.ts
-	cp -r $(TS_OUT)/segmented-cfmm/src/generated/ frontend/src/Generated/generated/
+	cp -a $(TS_OUT)/segmented-cfmm/src/generated frontend/src/Generated/
 
 # Quickly setup the frontend and run the it locally. Check `frontend/readme.md` for details.
-frontend:
+frontend: env = development
+frontend: testnet_link = https://granada.testnet.tezos.serokell.team/
+frontend: mainnet_link = https://mainnet.api.tez.ie/
+frontend: typescript
+	mkdir -p frontend/src/Generated
+
+	# Don't print error message if the file does not exist.
+	rm frontend/src/Generated/Env.ts 2> /dev/null || true
+
+	echo "export const env = \"$(env)\"" >> frontend/src/Generated/Env.ts
+	echo "export const testnetLink = \"$(testnet_link)\"" >> frontend/src/Generated/Env.ts
+	echo "export const mainnetLink = \"$(mainnet_link)\"" >> frontend/src/Generated/Env.ts
 	cd frontend && yarn && yarn tailwind && yarn dev
 
```

Comment:
I'm having a weird bug now...

If I select the "3 minutes" option, the contract will fail with error 103 (aka the `past_deadline_err` error).
If I manually select a time 3 minutes in the future from the datetime picker, the call works.

I recorded a short video below.

Even selecting the "1 day" option fails sometimes as well.

https://user-images.githubusercontent.com/718166/139691479-6245f9b4-24b7-42ab-886b-89289188d1cf.mp4

Reply:
That's weird, could you give me some detail on what entrypoint and parameters? Seems like the video is not working for me, with the error the file is corrupt.

Reply:
Just called the `swap x for y` entrypoint, with `x=10`, but I think this happens in other entrypoints too.

Here, I tried converting the video to another format, does this one work for you?

https://user-images.githubusercontent.com/718166/139812407-e1e15255-67b0-4676-a132-aec5c977acbc.mov